### ISSUE TITLE
Passing reference to other threads

### ISSUE THREAD

**Issue by eschnett** [bshoshany/thread-pool#6]
Line 108
```C++
            push_task([&start, &end, &loop, &blocks_running] {
```
passes references to `start` and `stop` to other threads. These variables are then overwritten in the next iteration, and may even go out of scope when the function finishes. This is undefined behaviour. Instead, the thread should receive copies of these variables.

**Comment by bshoshany** [comment]
Good catch! Thanks for spotting it. I will fix it in v1.3 (to be released tomorrow).

**Comment by bshoshany** [comment]
Closing since this issue is now fixed in v1.3.
